powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/encryptor/ecies/EciesRequestResponseValidator.java

```diff
@@ -65,6 +70,9 @@ public boolean validateEncryptedRequest(EncryptedRequest request) {
         if (request.getEphemeralPublicKey() == null || request.getEncryptedData() == null || request.getMac() == null) {
             return false;
         }
+        if (useTemporaryKeys == (request.getTemporaryKeyId() == null)) {
```

Comment:
So you want to forbid the temporary key for other versions, right?

Reply:
Yes.